File: dataset_reduzido/avatar/Python/Code/codeforces_203_A.py
xxs = input ( ).split ( " " )
x , t , a , b , da , db = int ( xxs [ 0 ] ) , int ( xxs [ 1 ] ) , int ( xxs [ 2 ] ) , int ( xxs [ 3 ] ) , int ( xxs [ 4 ] ) , int ( xxs [ 5 ] )
def main ( ) :
    if x == 0 :
        print ( "YES" )
        return "Result Found"
    a_time = min ( [ a // da , t - 1 ] )
    b_time = min ( [ b // db , t - 1 ] )
    for i in range ( a_time + 1 ) :
        for j in range ( b_time + 1 ) :
            if a - da * i == x or b - db * j == x :
                print ( "YES" )
                return "Result Found"
            if ( a - da * i ) + ( b - db * j ) == x :
                print ( "YES" )
                return "Result Found"
    return "Result Not Found"
ff = main ( )
if ff == "Result Not Found" :
    print ( "NO" )

File: output/Llama-2-100/avatar/Python/Java/codeforces_203_A.java
public class codeforces_203_A {
    public static void main(String[] args) {
        String xxs = input().split(" ");
        int x, t, a, b, da, db;
        x = Integer.parseInt(xxs[0]);
        t = Integer.parseInt(xxs[1]);
        a = Integer.parseInt(xxs[2]);
        b = Integer.parseInt(xxs[3]);
        da = Integer.parseInt(xxs[4]);
        db = Integer.parseInt(xxs[5]);
        if (x == 0) {
            System.out.println("YES");
            return "Result Found";
        }
        int aTime = Math.min(a / da, t - 1);
        int bTime = Math.min(b / db, t - 1);
        for (int i = 0; i < aTime + 1; i++) {
            for (int j = 0; j < bTime + 1; j++) {
                if (a - da * i == x || b - db * j == x) {
                    System.out.println("YES");
                    return "Result Found";
                }
                if ((a - da * i) + (b - db * j) == x) {
                    System.out.println("YES");
                    return "Result Found";
                }
            }
        }
        System.out.println("NO");
    }
}


